go/adbc/utils/utils.go

```diff
@@ -73,3 +81,28 @@ func removeFieldMetadata(field *arrow.Field) arrow.Field {
 		Metadata: arrow.Metadata{},
 	}
 }
+
```

Comment:
Since these are public facing, should we add docstrings comments for these? If they aren't intended to be externally facing, then we should move them to an `internal/` package.

Reply:
Moved functions to `internal/shared_utils.go` (and added docstrings). This means I had to remove instrumentation of `QueryContext` in `sqldriver/driver.go` - but I think each driver can instrument their own `ExecuteQuery` functions.